test/basic_unified_tests.jl

```diff
@@ -0,0 +1,198 @@
+
+        nbytes_read = get!(buffer, "test100B.csv", read_config)
+        @test nbytes_read == 20_000_000
+        @test String(buffer[1:nbytes_read]) == input
+    end
+end
+end # @testsetup
+
+@testitem "Basic BlobStorage usage" setup=[InitializeObjectStore, ReadWriteCases] begin
+using CloudBase.CloudTest: Azurite
+using RustyObjectStore: AzureConfig, ClientOptions
+
+
+# For interactive testing, use Azurite.run() instead of Azurite.with()
+# conf, p = Azurite.run(; debug=true, public=false); atexit(() -> kill(p))
+Azurite.with(; debug=true, public=false) do conf
+    _credentials, _container = conf
+    base_url = _container.baseurl
+    config = AzureConfig(;
+        storage_account_name=_credentials.auth.account,
+        container_name=_container.name,
+        storage_account_key=_credentials.auth.key,
+        host=base_url
+    )
+
+    run_read_write_test_cases(config)
+end # Azurite.with
+
+end # @testitem
+
+# NOTE: PUT on azure always requires credentials, while GET on public containers doesn't
+@testitem "Basic BlobStorage usage (anonymous read enabled)" setup=[InitializeObjectStore, ReadWriteCases] begin
+using CloudBase.CloudTest: Azurite
+using RustyObjectStore: AzureConfig, ClientOptions
+
+# For interactive testing, use Azurite.run() instead of Azurite.with()
+# conf, p = Azurite.run(; debug=true, public=true); atexit(() -> kill(p))
+Azurite.with(; debug=true, public=true) do conf
+    _credentials, _container = conf
+    base_url = _container.baseurl
+    config = AzureConfig(;
+        storage_account_name=_credentials.auth.account,
+        container_name=_container.name,
+        storage_account_key=_credentials.auth.key,
+        host=base_url
+    )
+    config_no_creds = AzureConfig(;
+        storage_account_name=_credentials.auth.account,
+        container_name=_container.name,
+        host=base_url
+    )
+
+    run_read_write_test_cases(config_no_creds, config)
+end # Azurite.with
+end # @testitem
+
+@testitem "Basic AWS S3 usage" setup=[InitializeObjectStore, ReadWriteCases] begin
+using CloudBase.CloudTest: Minio
+using RustyObjectStore: AwsConfig, ClientOptions
+
+# For interactive testing, use Minio.run() instead of Minio.with()
+# conf, p = Minio.run(; debug=true, public=false); atexit(() -> kill(p))
+Minio.with(; debug=true, public=false) do conf
+    _credentials, _container = conf
+    base_url = _container.baseurl
+    default_region = "us-east-1"
+    config = AwsConfig(;
+        region=default_region,
+        bucket_name=_container.name,
+        access_key_id=_credentials.access_key_id,
+        secret_access_key=_credentials.secret_access_key,
+        host=base_url
+    )
+
+    run_read_write_test_cases(config)
+
+end # Minio.with
+end # @testitem
+
+@testitem "Basic AWS S3 usage (anonymous read enabled)" setup=[InitializeObjectStore, ReadWriteCases] begin
+# TODO: currently object_store defaults to Instance credentials when no other credentials are supplied
```

Comment:
We would need some extra work to prevent object_store from trying to fetch the AWS instance credentials if no other credentials are provided. This would allow anonymous reads but I deemed it not worthy of our time currently as it is unlikely that we are going to use this in the near term. What do you think?

Reply:
ah, yeah that decision makes complete sense to me! i just didn't understand that from the comment, maybe let's copy-paste your explanation into an issue and link it here?